MCprep_addon/materials/generate.py

```diff
@@ -216,6 +217,32 @@ def find_from_texturepack(blockname: str, resource_folder: Optional[Path]=None)
 	return res
 
 
+def get_format_version_texturepack(resource_folder: Optional[Path]=None) -> Union[int, MCprepError]:
+	""" Get texturepack format version
+
+	See https://minecraft.wiki/w/Pack_format#List_of_resource_pack_formats"""
+	if resource_folder is None:
+		# default to internal pack
+		resource_folder = Path(cast(
+			str,
+			bpy.path.abspath(bpy.context.scene.mcprep_texturepack_path)
+		))
+
+	if not resource_folder.exists() or not resource_folder.is_dir():
+		env.log("Error, resource folder does not exist")
+		line, file = env.current_line_and_file()
+		return MCprepError(FileNotFoundError(), line, file, f"Resource pack folder at {resource_folder} does not exist!")
+
+	# Resource folder is same level as assets folder
+	file = Path(resource_folder, "pack.mcmeta")
+	if (file.is_file()):
+		with open(file, 'r') as f:
+			data = json.load(f)
+			return data["pack"]["pack_format"]
+	# return the unaffected change version, 22
+	return 21
```

Comment:
All `pack.mcmeta` files should define the pack format, so I wouldn't include this catch-all `return 21`. It could result in some silent errors.

I'd prefer an `MCprepError` object be returned if `pack.mcmeta` doesn't contain the pack format or if the file doesn't exist. To me at least, that's a bug with the resource pack.

A check for the `pack` and `pack_format` keys should also be added, just because there's no guarantee that either key will exist. I would have also suggested a `TypedDict` since `pack.mcmeta` has a well-defined structure, but I don't think Blender 2.8X has a new enough Python version for that.

Reply:
I was thinking of the current default MCprep resourcepack doesn't have the meta file and if you make it return like that this whole [check_dirs](https://github.com/Moo-Ack-Productions/MCprep/blob/d1eb1465f81a214ef9acb809f16ee7f79258db92/MCprep_addon/materials/generate.py#L162) part is considered as unnecessary, throw away the legacy behavior, you can swap any sub folder like from the "textures" folder can also work.